src/jdk.incubator.code/share/classes/jdk/incubator/code/dialect/core/CoreOp.java

```diff
@@ -318,6 +320,85 @@ public Block.Builder lower(Block.Builder b, CodeTransformer _ignore) {
             b.rebind(b.context(), CodeTransformer.LOWERING_TRANSFORMER).op(this);
             return b;
         }
+
+        static CoreOp.FuncOp convertInvokeToFuncCallOp(CoreOp.FuncOp funcOp, MethodHandles.Lookup l) {
+            return funcOp.transform(funcOp.funcName(), (blockBuilder, op) -> {
+                if (op instanceof JavaOp.InvokeOp iop) {
+                    Method invokeOpCalledMethod = null;
+                    try {
+                        invokeOpCalledMethod = iop.invokeDescriptor().resolveToMethod(l);
+                    } catch (ReflectiveOperationException e) {
+                        throw new RuntimeException(e);
+                    }
+                    if (invokeOpCalledMethod instanceof Method m) {
+                        CoreOp.FuncOp f = Op.ofMethod(m).orElse(null);
+                        if (f != null) {
+                            Op.Result result = blockBuilder.op(CoreOp.funcCall(
+                                    iop.invokeDescriptor().name(),
+                                    f.invokableType(),
+                                    blockBuilder.context().getValues(iop.operands())));
+                            blockBuilder.context().mapValue(op.result(), result);
+                            return blockBuilder;
+                        }
+                    }
+                }
+                blockBuilder.op(op);
+                return blockBuilder;
+            });
+        }
+
+        static CoreOp.FuncOp invokeToFuncOp(JavaOp.InvokeOp invokeOp, MethodHandles.Lookup l) {
+            try {
+                Method method = invokeOp.invokeDescriptor().resolveToMethod(l);
+                return Op.ofMethod(method).orElse(null);
+            } catch (ReflectiveOperationException e) {
+                throw new IllegalStateException("Could not resolve invokeOp to method");
```

Comment:
Just pass on `e` in the thrown exception in RuntimeException like in `convertInvokeToFuncCallOp`.